src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -57,6 +59,12 @@ public DocumentExecuter(IDocumentBuilder documentBuilder, IDocumentValidator doc
             _documentCache = documentCache ?? throw new ArgumentNullException(nameof(documentCache));
         }
 
+        public DocumentExecuter(IDocumentBuilder documentBuilder, IDocumentValidator documentValidator, IComplexityAnalyzer complexityAnalyzer, IDocumentCache documentCache, IEnumerable<IConfigureExecution>? configurations)
+            : this(documentBuilder, documentValidator, complexityAnalyzer, documentCache)
+        {
+            _configurations = configurations;
```

Comment:
`_configurations = configurations;` -> `_configurations = configurations.ToArray()/ToList();` + change field type

I always looked at the use of `IEnumerable<T>` with little distrust. In addition, remember the possible allocation of the iterator for each call.

Reply:
I don't know if it matters, but fine with me.